examples/key.tf

```diff
@@ -0,0 +1,8 @@
+resource "ably_api_key" "api_key_1" {
+  app_id = ably_app.app1.app_id
+  name   = "KeyName"
+  capability = {
+    channel1 = ["publish", "subscribe"]
```

Comment:
Just thinking about this myself - `channel1` and every other field in this object will have an arbitrary name, that complies with our [resource names and wildcards guidance](https://ably.com/docs/core-features/authentication#wildcards): 

> Capabilities are a map from resources to a list of [operations](https://ably.com/docs/core-features/authentication#capability-operations). Each resource can match a single channel e.g. channel, or multiple channels using wildcards (*). Wildcards can only replace whole segments (segments are delimited by :) of the resource name. A wildcard at the end of the name can replace arbitrarily many segments.

I am not convinced (I just don't know) wether Terraform syntax will allow all these possible options (especially wildcards) as valid object field names. 

If not - should we just turn `capability` into a JSON field instead?

Reply:
Looking into this: https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md#identifiers (based on `Identifier` being on the left side of the equal sign in [object expression](https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md#collection-values)), and specifically this: 

> The Unicode specification provides the normative requirements for identifier parsing. Non-normatively, the spirit of this specification is that ID_Start consists of Unicode letter and certain unambiguous punctuation tokens, while ID_Continue augments that set with Unicode digits, combining marks, etc.

I feel like it will be better in terms of not violating the "spirit" of the spec to turn `capability` into JSON.

Reply:
We could also provide a name parameter to account for arbitrary naming?

Json within HCL can get messy pretty quickly.  We could use jsonencode function but I'd rather we try to keep everything to HCL is possible.

Reply:
So something like 
```
capabilities = [
    { 
        resource = ""
        capability = ["publish", ...]
    },
    ...
]
```

?

Reply:
Yep, exactly. 
As a side note,  I would call the parameter "name" rather than "resource" but I don't have a very strong opinion on it.

Reply:
The reason I used “resource” is because that’s how we call these things in
the Ably documentation. (It is a variation on a channel name, but because
of wildcard support it is slightly different). Tag the same time I clearly
see how this can be confused with terraform resource…

Should we maybe call it “channel”?

I suggest we choose one option to move forward with now, but then also ask
DevEd team opinion, and potentially update/fix if needed in the future?

Reply:
And from @tbedford:

> On that specific question, yes resource is what they've always been referred to as, where a resource is some combination of namespace/channel. I suppose where it helped you could explicitly call out namespaces or channels.

I suggest we can move forward with "resource", and then get some early user feedback on wether that will be confusing or not.

Reply:
I've just returned to this one. Here's an alternate setup which could make things neater than nesting capabilities within the ably_api_key resource:

```terraform
resource "ably_api_key" "api_key_1" {
  app_id = ably_app.app1.app_id
  name   = "KeyName"
}

resource "ably_api_key_capability" "channel1" {
  api_key = ably_api_key.api_key_1.name
  resource = "channel1"
  capabilities = ["publish", "subscribe"]
}

resource "ably_api_key_capability" "channel2" {
  api_key = ably_api_key.api_key_1.name
  resource = "channel2"
  capabilities = ["history"]
}
```

Reply:
Looks great!

Reply:
Resolving as I think this solves the issue.